src/descriptors/analysis.rs

```diff
@@ -776,3 +775,72 @@ impl PartialSpendInfo {
         &self.recovery_paths
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use std::str::FromStr;
+
+    #[test]
+    fn valid_key() {
+        let xpub_str =
+            "[8c3ffb6e/48'/1'/0'/2']tpubDEMt3bpQMa99W81K9h8f2FJH1C81eSd6bbSkBP8tcqQHAfSKvuGp2fz6xiVpfShzT9sKPx7DVBphChjxvNd15WcbsCca5oVz1AcUTWHxkdS/<0;1>/*";
+        let key = descriptor::DescriptorPublicKey::from_str(xpub_str).unwrap();
+        let mut checker = DescKeyChecker::new();
+        assert!(checker.check(&key).is_ok());
+    }
+
+    #[test]
+    fn invalid_key() {
+        // Multipath of size 3
+        let xpub_str =
+            "[8c3ffb6e/48'/1'/0'/2']tpubDEMt3bpQMa99W81K9h8f2FJH1C81eSd6bbSkBP8tcqQHAfSKvuGp2fz6xiVpfShzT9sKPx7DVBphChjxvNd15WcbsCca5oVz1AcUTWHxkdS/<0;1;2>/*";
+        let key = descriptor::DescriptorPublicKey::from_str(xpub_str).unwrap();
+        let mut checker = DescKeyChecker::new();
+        assert!(matches!(
+            checker.check(&key),
+            Err(LianaPolicyError::InvalidKey(_))
```

Comment:
nit: I don't think you need to change it, but I was just thinking that to be sure the error contains `key`, perhaps this could be changed to:
```
assert!(matches!(
    checker.check(&key),
    Err(LianaPolicyError::InvalidKey(k)) if k == key.into()
));
```
I'm not sure if this is best practice, but it just occurred to me while reviewing.